src/main/scala/io/iohk/ethereum/rlp/package.scala

```diff
@@ -0,0 +1,36 @@
+package io.iohk.ethereum
+
+package object rlp {
+
+  case class RLPException(message: String) extends RuntimeException(message)
+
+  sealed trait RLPEncodeable
+
+  case class RLPList(items: RLPEncodeable*) extends RLPEncodeable
+
+  case class RLPValue(bytes: Array[Byte]) extends RLPEncodeable
+
+  trait RLPEncoder[T] {
+    def encode(obj: T): RLPEncodeable
+  }
+
+  trait RLPDecoder[T] {
+    def decode(rlp: RLPEncodeable): T
+  }
+
+  def encode[T](input: T)(implicit enc: RLPEncoder[T]): Array[Byte] = RLP.encode(enc.encode(input))
```

Comment:
why do we need these methods? Can't we just have `def encode[T](input: T)(implicit enc: RLPEncoder[T])` in `RLP` obj?